### DISCUSSION TITLE
Capitolo 5 - Stringa contenenti le classi non identica

### DISCUSSION THREAD

**Discussion by NersOfficial** [lifeisfoo/react-guida-pratica#5] [Errori e refusi]
Durante i test dei codici ho riscontrato un ulteriore errore che coinvolge la costante denominata in modo errato nel listato 5.3 ( [qui per sapere di più](https://github.com/lifeisfoo/react-guida-pratica/discussions/4) ).

Il contenuto di questa è diverso tra il listato 5.3 e 5.5, nello specifico la prima contiene meno classi rispetto alla seconda. 

Ecco il codice della costante nello specifico:

`Listato 5.3`
```jsx
>> const itemClasses = "list-group-item border rounded-1 my-1";
```

`Listato 5.5`
```jsx
>> const todoClasses = "d-flex align-items-center list-group-item my-1 border rounded-1"; 
```

Escludendo il nome, la seconda contiene più classi rispetto alla prima. Controllando nel testo (sia cartaceo che digitale), le classi ` d-flex align-items-center ` non sono specificate e quindi manca una loro spiegazione (come fatto nelle altre classi boostrap incontrate in precedenza) e nel `Listato 5.5` non è evidenziata come modifica (in grassetto come fatto per ogni modifica).

**Comment by lifeisfoo** [comment]
C'è stato sicuramente qualche problema durante l'impaginazione è il grassetto è andato perduto. La spiegazione, invece, ho proprio dimenticato di inserirla: la trovi di seguito con anche delle immagini che spero rendano tutto più chiaro.

La classe [`d-flex`](https://getbootstrap.com/docs/5.1/utilities/flex/) aggiunta la proprietà `display: flexbox` all'elemento, e permette quindi di posizionare correttamente gli altri elementi al suo interno.

![image](https://user-images.githubusercontent.com/3725610/230640869-7fbc57ae-d63f-4839-a508-3ff2686f730f.png)

Se non avessimo usato anche la classe [`align-items-center`](https://getbootstrap.com/docs/5.1/utilities/flex/#align-items), gli elementi all'interno del contenitore con classe `d-flex` si sarebbero posizionati in modo non allineato rispetto all'asse delle y. Nota come la checkbox è leggermente più in alto rispetto al testo.

![image](https://user-images.githubusercontent.com/3725610/230640994-8f6149f3-39e2-4c30-af78-23514061c198.png)

Queste sono le varie possibile configurazioni verticali, il quarto esempio è quello con `align-items-center`.

![image](https://user-images.githubusercontent.com/3725610/230643620-1a99d26e-2d81-40d0-ae14-14c7ba6662b6.png)

Infine, se non avessimo usato la classe `d-flex`, gli elementi si sarebbero posizioni in modo non visivamente ordinato, come già abbiato detto prima. 

![image](https://user-images.githubusercontent.com/3725610/230641047-903087a3-1ed0-47f5-9d6f-bea457d6d9a5.png)

Spero che la spiegazione sia stata chiara.

Grazie @NersOfficial per la segnalazione :+1:
